TaskController

mirinion opened the pull request.

mirinion committed:
Написал контроллер и сервис для таск, поправил контроллер проектов.

alekseiiagn commented on src/main/java/ru/mephi/tasuku/appuser/dto/AppUserResponse.java:
```diff
@@ -14,8 +14,4 @@ public class AppUserResponse {
 	private long id;
 	private String username;
 	private String email;
-	private SystemRole systemRole;
```
роль нам как то мешала?

mirinion replied:
учел

alekseiiagn commented on src/main/java/ru/mephi/tasuku/binding/service/ProjectUserRoleService.java:
```diff
@@ -0,0 +1,17 @@
+import org.springframework.stereotype.Component;
+import ru.mephi.tasuku.binding.repository.ProjectUserRoleRepository;
+
+@Component
```
Сервисы помечаются аннотацией Service

mirinion replied:
учел

alekseiiagn commented on src/main/java/ru/mephi/tasuku/project/controller/ProjectController.java:
```diff
@@ -3,43 +3,49 @@
-
     private final ProjectService projectService;
-    private final ProjectDtoMapper mapper;
+    private final TaskService taskService;
```
стараемся всю бизнес логику переносить в service слой, тут должен быть подключен только сервис проекта

alekseiiagn replied:
возможно /{projectId}/tasks можно перенести в контроллер тасок

mirinion replied:
Но контроллер тасок настроен на request mapping ("/tasks"). Хочешь, чтобы ручка выглядела "/tasks/{projectId}/tasks"?

alekseiiagn replied:
резонно. Тогда давай сделаем в сервисе проекта метод getTasks или что то подобное, и там уже вычислим?

alekseiiagn commented on src/main/java/ru/mephi/tasuku/project/controller/ProjectDtoMapper.java:
```diff
@@ -18,6 +23,15 @@ public ProjectResponse objectToDto(Project object) {
+        return Project.builder()
+                .name(dto.getName())
+                .headUser(appUserService.findById(dto.getHeadUserId()))
+                .closed(false)
```
почему в этом классе мы захардкодили возврат false?

mirinion replied:
учел

alekseiiagn commented on src/main/java/ru/mephi/tasuku/project/service/ProjectService.java:
```diff
@@ -16,13 +16,13 @@ public class ProjectService {
     private final ProjectRepository projectRepository;
 
-    public Project getById(long id) {
+    public Project findById(long id) {
```
в сервисах обычно методы такие это get, а вот в репозиторий мы можем уже лезть по find или по get.
Считаю лишним такое переименование

mirinion replied:
учел

alekseiiagn commented on src/main/java/ru/mephi/tasuku/project/service/ProjectService.java:
```diff
@@ -37,7 +37,7 @@ public long createProject(Project project) {
     public void deleteProject(long id) {
         ProjectModel model = projectRepository.findById(id)
-                .orElseThrow(() -> new ProjectNotFoundException(id));
+                .orElseThrow(() -> new ProjectIdNotFoundException(id));
```
переименуй подобные эксепшены тогда уж в ProjectByIdNotFoundException

mirinion replied:
почему? "не найден айдишник проекта" = projectItNotFound

alekseiiagn commented on src/main/java/ru/mephi/tasuku/project/service/exception/FieldValidationException.java:
```diff
@@ -1,11 +1,9 @@
 public class FieldValidationException extends ProjectException {
     public FieldValidationException(String field) {
-        super(HttpStatus.BAD_REQUEST, ErrorMessage.VALIDATION_ERROR, field);
+        super(HttpStatus.BAD_REQUEST, ProjectErrorMessage.VALIDATION_ERROR, field);
```
возможно нам хватит только 1го класса ErrorMessage? В него модно добавить все строки, которые хотим юзать

mirinion replied:
ИМХО лучше разделить текста сообщений в разных энамах для разных сущностей. Так легче найти нужный текст ошибки.

alekseiiagn commented on src/main/java/ru/mephi/tasuku/project/service/exception/ProjectErrorMessage.java:
```diff
@@ -1,14 +1,14 @@
 
-    PROJECT_NOT_FOUND("Project with 'projectId' = '%s' not found."),
-    PROJECT_EXISTS("Project with 'projectId' = '%s' already exists."),
+    PROJECT_ID_NOT_FOUND("Project with 'projectId' = '%s' not found."),
```
PROJECT_BY_ID_NOT_FOUND
e t c

mirinion replied:
аналогично: "не найден айдишник проекта" = projectItNotFound

alekseiiagn commented on src/main/java/ru/mephi/tasuku/security/SecurityConfig.java:
```diff
@@ -17,12 +17,15 @@ public class SecurityConfig {
-		return http.httpBasic()
+		http
+				.cors()
+				.and().csrf().disable()
+				.httpBasic()
 				.and()
 					.authorizeHttpRequests()
 					.anyRequest().authenticated()
-				.and().formLogin()
-				.and().build();
+				.and().formLogin();
```
выглядит страшно, попробуй начать с http.cors() и потом уже через enter, должно стать лучше

mirinion replied:
учел

alekseiiagn commented on src/main/java/ru/mephi/tasuku/security/UserConditionEvaluator.java:
```diff
@@ -3,20 +3,32 @@
-				.getPrincipal();
-		long userId = user.getAppUser().getId();
-		var projectUserRoles = projectUserRoleRepository
-				.searchAllByProjectIdAndUserId(userId, projectId);
-		return !projectUserRoles.isEmpty();
+				.getPrincipal()).getAppUser();
```
ощущение, что это общая часть, возможно стоит вынести в метод отдельный

mirinion replied:
учел

alekseiiagn commented on src/main/java/ru/mephi/tasuku/task/controller/dto/TaskCreateRequest.java:
```diff
@@ -0,0 +1,12 @@
+import lombok.Getter;
+
+@Getter
+public class TaskCreateRequest {
```
что, даже без конструктора?

mirinion replied:
учел

alekseiiagn commented on src/main/java/ru/mephi/tasuku/task/controller/dto/TaskUpdateRequest.java:
```diff
@@ -0,0 +1,14 @@
+import ru.mephi.tasuku.task.repository.model.TaskStatus;
+
+@Getter
+public class TaskUpdateRequest {
```
тоже без конструктора

mirinion replied:
учел

alekseiiagn commented on src/main/java/ru/mephi/tasuku/task/service/TaskModelMapper.java:
```diff
@@ -14,27 +14,36 @@
 
 	public static Task modelToObject(TaskModel model) {
-		return Task.builder()
+		var object = Task.builder()
```
давай не использовать var. Либо использовать его везде

mirinion replied:
учел

alekseiiagn commented on src/main/java/ru/mephi/tasuku/task/service/TaskService.java:
```diff
@@ -0,0 +1,80 @@
+
+	public List<Task> findAllInProject(long projectId) {
+		return taskRepository.getAllByProjectId(projectId)
+				.stream()
```
обычно .stream() не выносят на отдельную строку

mirinion replied:
учел

alekseiiagn commented on src/main/java/ru/mephi/tasuku/task/service/TaskService.java:
```diff
@@ -0,0 +1,80 @@
+		return taskRepository.save(taskModel).getId();
+	}
+
+	public void deleteTask(long taskId) {
```
все методы, обращающиеся в бд должны быть Transactional

mirinion replied:
учел

mirinion committed:
добавил роль в респонс юзера
ProjectUserRoleService пометил как @Service
Убрал захардкоженный closed=false из ProjectDtoMapper
Изменил в сервисах нейминг findBy на getBy
вернул @Data в request dto
повесил @Transactional на deleteTask

mirinion committed:
Поправил нейминг исключений

mirinion committed:
добавил getTasks в ProjectService

alekseiiagn reviewed (approved):
надеюсь, что все ок, если что, потом исправлю)
